src/common/Modal.tsx

```diff
@@ -0,0 +1,79 @@
+import React, {useState} from "react";
+import ReactDOM from 'react-dom';
+import Modal from 'react-modal';
+import "./Modal.css"
+
+const customStyles = {
+    content : {
+        top: '50%',
+        left: '50%',
+        right: 'auto',
+        bottom: 'auto',
+        transform: 'translate(-50%, -50%)',
+        width: '600px',
+        height: '400px',
+        border: 'solid 1px #707070',
+        backgroundColor: '#ffffff',
+        borderRadius: '18px',
+        boxShadow: '10px 10px 6px 0 rgba(0, 0, 0, 0.16)'
+}
+
+};
+
+
+interface Props {
+    title: String,
+    text: String,
+    onClick : ()=>void,
+    button: JSX.Element
+}
+
+Modal.setAppElement('body')
+
+
+const ModalComponent = (props: Props) => {
+
+    const [modalIsOpen,setIsOpen] = React.useState(false);
+    function openModal() {
+        setIsOpen(true);
+    }
+
+    function afterOpenModal() {
+    }
+
+    function closeModal(){
+        setIsOpen(false);
+    }
+
+    function confirmModal(){
+        props.onClick();
+        setIsOpen(false);
+    }
+
+    return (
+        <div>
+            <div onClick={openModal}>
+                {props.button}
```

Comment:
Again the modal should not render a button, the button should be a simple button in the component which uses this modal.